lib/elixir/lib/kernel.ex

```diff
@@ -1971,10 +1971,17 @@ defmodule Kernel do
   In case any entry returns `nil`, its key will be removed
   and the deletion will be considered a success.
   """
-  @spec pop_in(Access.t, nonempty_list(term)) :: {term, Access.t}
+  @spec pop_in(Access.container, nonempty_list(term)) :: {term, Access.container}
   def pop_in(data, keys)
-  def pop_in(nil, [h | _]), do: Access.pop(nil, h)
-  def pop_in(data, keys), do: do_pop_in(data, keys)
+
+  def pop_in(nil, [head | _]) do
+    raise ArgumentError,
```

Comment:
Why this change? Now we have the same logic in two different places. It is ok to call `Access.pop(nil, h)` to error.